src/leaking.md

```diff
@@ -104,22 +205,49 @@ mem::forget us in the middle of the iteration, all that does is *leak even more*
 (and possibly leave the Vec in an unexpected but otherwise consistent state).
 Since we've accepted that mem::forget is safe, this is definitely safe. We call
 leaks causing more leaks a *leak amplification*.
+-->
+
+ではどうすればいいのでしょうか? うーん、ちょっと一貫性のある状態を選択することが出来ます。
+すなわち、イテレーションの初めでは Vec の len を 0 に設定し、そしてもし必要ならば、
+デストラクタ内で len を修正します。このようにすることで、もしすべてが普通に実行されるなら、
+最小限のオーバーヘッドで望まれている振る舞いを得ることが出来ます。
+しかし、もし*大胆にも* mem::forget がイテレーションの真ん中に存在したら、
+この関数によって、*更に多くのものがリークされます* (そして多分 Vec を
+予期しない状態か、そうでないなら一貫性のある状態にするでしょう) 。 mem::forget は安全だとして
```

Comment:
Vec を、一貫性はあるとしてもユーザーが予期しない状態にするでしょう

when `Drain` is dropped after consuming a few elements without running its destructor, the `Vec`'s len remains 0, which is consistent (wrt `Vec`), because you cannot access any elements, but is unexpected by the users because `Drain` consumed only a few elements.

.... is what I think this paragraph explains.

Reply:
:+1: